### PR TITLE
fix(tag): Add listItem prop to tags ; update tag examples with UL wrapper

### PR DESCRIPTION
**Pull request by dakahn**
references #2738 

For screen reader users tags should be displayed in an unordered list and enumerated as list items. This PR enhances our example and adds functionality to tags so they can remain `<span>` elements, but have the aria-role `listitems`. 👍 

#### Changelog

\+ `<ul>` wrapper with descriptive role around tag examples
\+ `listItem` prop to Tag component that when set to true lets the tags be enumerated in a list for better 
screen reader compatibility.

#### Testing
You can add multiple tags within the `<ul>` and make the `<ul>` a tabstop by adding `tabIndex="0"`. Then turn on a screen reader and tab to the tag list.

### CODE REVIEW

File: packages/react/src/components/Tag/Tag.js [hunk 1]
```diff
@@ -36,13 +44,14 @@ const Tag = ({ children, className, type, filter, disabled, ...other }) => {
     <span
       className={tagClasses}
       title="Clear filter"
+      role={listItem && 'listitem'}
```
Comment (joshblack) [inline packages/react/src/components/Tag/Tag.js]:
```suggestion
      role={listItem ? 'listitem' : undefined}
```

Would it need to be this?

**Review by emyarod — COMMENTED** [review]
I was under the impression that individual tags could exist outside of a list, so #3017 makes the change to the docs only

but do we really want to introduce a new prop when we are already spreading props into the component?

**Comment by snidersd** [comment]
Retested with DAP, VO on MacOS 12.4 and NVDA on Windows 10 with no violations.

**Comment by dakahn** [comment]
@emyarod Yep! Looking at the code it seemed to built around the idea that tags exist _anywhere_. What we're doing here is two fold:
First, we're providing a proper usage example in the documentation. If you have tags, they should be enumerated in a list and that list should have  role that describes what the tags are all about. As far as I know tags _outside_ of a `<ul>` wont throw any accessibility violations in DAP or the like, but if you have three tags, lets say in a `<div>` those tags will be read like one continuous sentence. Which would fail an AVT3 check. 

Secondly, we're avoiding breaking changes (I hope) by maintaining functionality and usage expectations _out of the box_. But if you see our recommended usage (tags as list items, inside an unordered list element) you can -- and according to our forthcoming spec should -- give tags that list item role and throw em in an unordered list.

We're introducing a new prop here so that tags can be accessible. I'm cool with that. They're relatively simple anyway.

**Comment by dakahn** [comment]
Totally didn't see all the work you'd this afternoon @emyarod we can close this in favor of your solution and I'll make a note to add usage docs to the spec. 🥇

Guess we were just grinding on this at the same time 🤦‍♂

**Pull request closed without merging**